src/Containers.jsx

```diff
@@ -176,12 +177,12 @@ class Containers extends React.Component {
         const columnTitles = [_("Name"), _("Image"), _("Command"), _("CPU"), _("Memory"), _("State")];
         // TODO: emptyCaption
         let emptyCaption = _("No running containers");
-        const renderRow = this.renderRow;
         const containersStats = this.props.containersStats;
         // TODO: check filter text
-        let filtered = this.props.containers.filter(container => (!this.props.onlyShowRunning || container.State.Running));
-        let rows = filtered.map(function (container) {
-            return renderRow(containersStats, container);
+        let filtered = [];
+        Object.keys(this.props.containers).filter(id => { if (!this.props.onlyShowRunning || this.props.containers[id].State.Running) { filtered[id] = this.props.containers[id] } });
+        let rows = Object.keys(filtered).map(function (id) {
+            return this.renderRow(containersStats, this.props.containers[id]);
         }, this);
```

Comment:
Replace the `function` with an arrow, drop the `this` argument. Also, `filtered` is already a list, no need for `Object.keys` (and you set `filtered` elements as object keys again, which is inconsistent) - it should be a list as you want to show it as a table, and almost certainly want to sort it, too.

Reply:
here ```filtered``` and ```this.props.containers``` are map-like array, ```filtered[id] = this.props.containers[id]```. That's why ```Object.key``` be use. I'd better keep  ```this.props.containers``` as a map and keep ```filtered``` as an array?